### PR TITLE
Feat/extrac ingest v2

### PR DESCRIPTION
**Pull request by Tomvbe**

### CODE REVIEW

File: ldes-server-port-ingest-rest/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    <properties>
+        <relative-formatter-path>..</relative-formatter-path>
+    </properties>
+
+    <dependencies>
+        <!-- VSDS -->
+        <dependency>
+            <groupId>be.vlaanderen.informatievlaanderen.vsds</groupId>
+            <artifactId>ldes-server-domain</artifactId>
```
Comment (Tomvbe) [inline ldes-server-port-ingest-rest/pom.xml:23]: This is still used for for configuration (appconfig) and some util functions. We have VSDSPUB-662 to figure this out.

File: ldes-server-infra-mongo/mongo-ingest-repository/pom.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    <artifactId>mongo-ingest-repository</artifactId>
+
+    <dependencies>
+
+<!--        VSDS-->
+        <dependency>
+            <groupId>be.vlaanderen.informatievlaanderen.vsds</groupId>
+            <artifactId>ldes-server-port-ingest</artifactId>
+            <version>0.0.9-SNAPSHOT</version>
```
Comment (WLefever-Cegeka) [inline ldes-server-infra-mongo/mongo-ingest-repository/pom.xml:20]: project.version

File: ldes-server-infra-mongo/mongo-ingest-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/mapper/MemberEntityMapper.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFDataMgr;
+import org.apache.jena.riot.RDFParserBuilder;
+import org.springframework.stereotype.Component;
+
+import java.io.StringWriter;
+
+@Component
+public class MemberEntityMapper {
```
Comment (WLefever-Cegeka) [inline ldes-server-infra-mongo/mongo-ingest-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/mapper/MemberEntityMapper.java:14]: Would personally avoid the Component annotation here (is easier if later on somebody creates an integration test for MemberRepositoryImpl). And just create an instance where this is used.

File: ldes-server-port-ingest-rest/pom.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        </dependency>
+        <dependency>
+            <groupId>be.vlaanderen.informatievlaanderen.vsds</groupId>
+            <artifactId>ldes-server-port-ingest</artifactId>
+            <version>0.0.9-SNAPSHOT</version>
```
Comment (WLefever-Cegeka) [inline ldes-server-port-ingest-rest/pom.xml:28]: project.version

File: ldes-server-port-ingest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImplTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+class MemberIngesterImplTest {
+
+	@Mock
+	private MemberRepository memberRepository;
+
+	@Mock
+	private ApplicationEventPublisher eventPublisher;
+
+	@InjectMocks
+	private MemberIngesterImpl memberIngestService;
```
Comment (WLefever-Cegeka) [inline ldes-server-port-ingest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImplTest.java:38]: would personally do this without the extension (speeds up the test as well), but to be discussed in the developer meeting (I heard Jonas already sent this topic :-) )

File: ldes-server-port-ingest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImplTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+		when(memberRepository.memberExists(member.getId())).thenReturn(true);
+
+		memberIngestService.ingest(member);
+
+		InOrder inOrder = inOrder(memberRepository, eventPublisher);
+		inOrder.verify(memberRepository,
+				times(1)).memberExists(member.getId());
+		inOrder.verifyNoMoreInteractions();
+		verifyNoInteractions(eventPublisher);
+		verifyNoMoreInteractions(memberRepository);
```
Comment (WLefever-Cegeka) [inline ldes-server-port-ingest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImplTest.java:57]: these are redundant

**Review by WLefever-Cegeka — APPROVED** [review]
Hi Tom,
Thanks for the nice work!
Made a couple of remarks, rather 'naar vorm dan naar inhoud' :-)

File: ldes-server-port-ingest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImplTest.java (hunk 5 above)
Comment (Tomvbe) [reply ldes-server-port-ingest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImplTest.java:38]: I added this to the agenda of the dev meeting :+1:

File: ldes-server-port-ingest-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/rest/config/IngestRestConfig.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.ingest.rest.config;
+
+import jakarta.annotation.PostConstruct;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFParser;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class IngestRestConfig {
```
[the remark this answers is not in the document]
Comment (Tomvbe) [reply ldes-server-port-ingest-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/rest/config/IngestRestConfig.java:9]: I am not sure about this. You could use custom fragmentations or just pagination with members that contain geospatial properties. Then the geospatial fragmenter would not be active and this bean would not be started.

File: ldes-server-port-ingest-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/rest/config/IngestRestConfigTest.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+class IngestRestConfigTest {
+
+	@Test
+	void triggerEspgDatabaseInitializationOnStartupShouldNotFail() {
+		assertDoesNotThrow(() -> new IngestRestConfig().triggerEspgDatabaseInitializationOnStartup());
+	}
```
[the remark this answers is not in the document]
Comment (Tomvbe) [reply ldes-server-port-ingest-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/rest/config/IngestRestConfigTest.java:12]: to be discussed in reviewcomment IngestRestConfig

File: ldes-server-infra-mongo/mongo-ingest-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/mapper/MemberEntityMapper.java (hunk 3 above)
Comment (Tomvbe) [reply ldes-server-infra-mongo/mongo-ingest-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/mapper/MemberEntityMapper.java:14]: To be discussed. I don't agree with this statement. Dependency injection would make testing easier and not harder. It also removes prevents the classes using this mapper (currently only MemberRepositoryImpl) not to require any information on how the mapper works, what it needs or how it needs to be created. If/when we need to add a dependency to the mapper (for example to map a new field), we would have to refactor everywhere the mapper is used and also figure out how to get that dependency in there. The MemberRepositoryImpl would become tightly coupled with the mapper. It is possible we may want to use this mapper on other locations as well, augmenting the problem. All the advantages of using DI apply here.

**Pull request merged**